.github/workflows/ci-build-and-smoke-fixed.yml

```
@@ -0,0 +1,84 @@
+  build:
+    name: Build and Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: 'false'
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v3
```

actionlint (1.7.7)

the runner of "actions/setup-dotnet@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)